lib/galaxy/tools/wrappers.py

```diff
@@ -80,14 +80,16 @@ def __init__(self, input, value, other_values={}):
         self._other_values = other_values
 
     def __eq__(self, other):
-        if isinstance(other, string_types):
-            return str(self) == other
-        elif isinstance(other, int):
-            return int(self) == other
-        elif isinstance(other, float):
-            return float(self) == other
-        else:
-            return super(InputValueWrapper, self) == other
+        return self.get_cast_value() == other
+
+    def get_cast_value(self):
```

Comment:
```suggestion
    def _get_cast_value(self):
```
unless you intended this to be exposed.

Also I'd move it before `__eq__()`.